sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -45,7 +69,326 @@ protected boolean isEnabled() {
     @Override
     protected List<Violation> _run(
             GraphTraversalSource g, GraphTraversal<Vertex, Vertex> eligibleVertices) {
-        return new ArrayList<>();
+        reset(g);
+        List<MethodVertex> candidateVertices = getCandidateVertices(eligibleVertices);
+        seekMethodUsages(candidateVertices);
+        return convertMethodsToViolations();
+    }
+
+    private void reset(GraphTraversalSource g) {
```

Comment:
State needs to be reset at the start of each execution, or else tests fail.

Reply:
While this works for the current iteration, we may want to retain the state when we grow to cross-class checks. In fact, we could load each class and immediately check for its private methods usage. Once we have all the classes loaded to the graph, we could run queries across multiple classes.